docs/proposals/service-discovery/README.md

```diff
@@ -159,40 +118,208 @@ proposal will be implemented, this is the place to discuss them.
 
 ### API changes
 
-The optimization design for the MultiClusterService API needs to be further iterated and improved, such as fixing the annotation `discovery.karmada.io/strategy` in the spec.
+This proposal proposes two new fields `ServerLocaltionClusters` and `ClientLocationClusters` in the `MultiClusterService` API.
+
+```go
+type MultiClusterService struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the desired state of the MultiClusterService.
+	Spec MultiClusterServiceSpec `json:"spec"`
+
+	// Status is the current state of the MultiClusterService.
+	// +optional
+	Status corev1.ServiceStatus `json:"status,omitempty"`
+}
+
+type MultiClusterServiceSpec struct {
+	// Types specifies how to expose the service referencing by this
+	// MultiClusterService.
+	// +required
+	Types []ExposureType `json:"types"`
+
+	// Ports is the list of ports that are exposed by this MultiClusterService.
+	// No specified port will be filtered out during the service
+	// exposure and discovery process.
+	// All ports in the referencing service will be exposed by default.
+	// +optional
+	Ports []ExposurePort `json:"ports,omitempty"`
+
+	// ServerLocaltionClusters specifies the clusters which will provide the service backend.
+	// If leave it empty, we will collect the backend pods' IP from all clusters and sync them to the ClientLocationClusters.
+	// +optional
+	ServerLocaltionClusters []string `json:"serverLocationClusters,omitempty"`
+
+	// ClientLocationClusters specifies the clusters which will request the service.
+	// If leave it empty, the service will be exposed to all clusters.
+	// +optional
+	ClientLocationClusters []string `json:"clientLocationClusters,omitempty"`
+}
+
+// ExposureType describes how to expose the service.
+type ExposureType string
+
+const (
+	// ExposureTypeCrossCluster means a service will be accessible across clusters.
+	ExposureTypeCrossCluster ExposureType = "CrossCluster"
+
+	// ExposureTypeLoadBalancer means a service will be exposed via an external
+	// load balancer.
+	ExposureTypeLoadBalancer ExposureType = "LoadBalancer"
+)
+```
 
-### General Idea
+With this API, we will:
+* Use `ServerLocationClusters` to specify the member clusters for propagating services from the Karmada control plane.
+* Use `ClientLocationClusters` to specify the `EndpointSlice` resources for synchronization from `ServerLocationClusters`.
```

Comment:
Do we need to synchronize the Service to the Client clusters?

Reply:
Yes, if we want to access the service in client clusters, we need to create it.